### PR TITLE
Add AsOfMatchRule and add match rule param in aj

### LINKED ISSUE #2240: python aj missing match rule option
With an `aj`, the user should be able to specify LT (less than) or LTE (less than or equal) (by default it's LTE).

Similarly with `raj`.

### PR DESCRIPTION
**Pull request by jmao-denver**
Fixes #2240

### CODE REVIEW

File: pyintegration/tests/test_table.py [hunk 1]
```diff
@@ -289,15 +274,15 @@ def test_as_of_join(self):
             result_table = left_table.aj(right_table, on=["a"])
             self.assertGreater(result_table.size, 0)
             self.assertLessEqual(result_table.size, left_table.size)
-            result_table = left_table.aj(right_table, on="a")
+            result_table = left_table.aj(right_table, on="a", joins="e", match_rule=AsOfMatchRule.LESS_THAN)
```
Comment (devinrsmith) [inline pyintegration/tests/test_table.py:277]: This test should still pass without setting this param, right? The default is LTE.

File: pyintegration/tests/test_table.py [hunk 2]
```diff
@@ -289,15 +274,15 @@ def test_as_of_join(self):
...
             self.assertGreater(result_table.size, 0)
             self.assertLessEqual(result_table.size, left_table.size)
 
         with self.subTest("reverse-as-of join"):
             result_table = left_table.raj(right_table, on=["a"])
             self.assertGreater(result_table.size, 0)
             self.assertLessEqual(result_table.size, left_table.size)
-            result_table = left_table.raj(right_table, on="a")
+            result_table = left_table.raj(right_table, on="a", joins="e", match_rule=AsOfMatchRule.GREATER_THAN)
```
Comment (devinrsmith) [inline pyintegration/tests/test_table.py:285]: ditto

File: pyintegration/tests/test_table.py (hunk 1 above)
Comment (jmao-denver) [reply pyintegration/tests/test_table.py:277]: The modified case tests both if a single string value for 'on' works and if aj can accept the non-default match rule.

File: pyintegration/tests/test_table.py (hunk 2 above)
Comment (jmao-denver) [reply pyintegration/tests/test_table.py:285]: see above.

**Review by devinrsmith — APPROVED** [review]

**Pull request merged**